### ISSUE TITLE
Worked best last month, but today not work

### ISSUE THREAD

**Issue by partizanfm88** [MarkSnaile/telegram-channel-views-boost#12]
It worked best for me for last months. Of course I have changed the code a little bit and it worked very good. But today stopped. Run the script at the same time on 2 VPS's and on PC and views not increase. Seems like telegram has changed something.

**Comment by Dartrisen** [comment]
Same problem btw. It had worked perfectly with my modifications. Really seems like telegram has changed something. Fetching is fine, proxy is fine but it can't add some views. If you solve it please respond.

**Comment by partizanfm88** [comment]
Yes, it returns key it send key back it receives "TRUE", the proxy are ok. Everything is good but views not increase. I have also tried to buy some views from services in internet and I got views. Looks like not all the methods were banned. But maybe these services use bots or real accounts. Anyway it would be nice to find out how to fix it.

**Comment by farrukhchik** [comment]
@partizanfm88  Can you provide proxies.txt example file? I am getting such kind of error
![image](https://user-images.githubusercontent.com/17916606/56362665-dcc78d00-6203-11e9-89d0-57d20bcd8081.png)

**Comment by partizanfm88** [comment]
> @partizanfm88 Can you provide proxies.txt example file? I am getting such kind of error
> ![image](https://user-images.githubusercontent.com/17916606/56362665-dcc78d00-6203-11e9-89d0-57d20bcd8081.png)

Your error not about proxy. And this script no longer works

**Comment by fredericofeliz** [comment]
@partizanfm88 did you fix the script? please tell me.

**Comment by Dartrisen** [comment]
I don't think you could fix it. It seems that we should add unique id number to a get request when you're trying to add a view. It is obviously you can't add a view from person who isn't use telegram.

**Comment by KingMatrix1989** [comment]
It seems with `url = https://t.me/v/?views= + key`  in `addViewToPost` function, the problem will be fixed!

**Comment by Dartrisen** [comment]
You're right. I had checked it out and it's working now. Thank you!

**Comment by sohail213** [comment]
I don't understand pls help with what is wrong with this code I am getting error proxy terminated

**Comment by sohail213** [comment]
Thread with proxy 95.167.177.18:8080 has been terminated. This error

**Comment by Dartrisen** [comment]
This is not an error.

**Comment by majidfarzaneh** [comment]
it does not work for me
whats the problem?
@Dartrisen can you tel me your code?

**Comment by majidfarzaneh** [comment]
@Dartrisen the get result return to me html of the link result but the post view its same to before

**Comment by jburagev** [comment]
Hi, is this script still working? i got success in the code but my views are not increasing... @skmohammadi can you please explain more in details what do you mean by "It seems with url = https://t.me/v/?views= + key in addViewToPost function, the problem will be fixed!" ? 

Thanks

**Comment by khorammfar** [comment]
**Thats Not Working**

**Comment by KingMatrix1989** [comment]
@jburagev,
try:
```
r = requests.request("GET", url, params=querystring, proxies={'https':proxy})
cookie = r.headers['set-cookie'].split(';')[0]
key = r.text.split('data-view="')[1].split('"')[0]
if 'stel_ssid' in cookie: 
  return {'key':key,'cookie':cookie}
else:
  return False
```
after getting *key* and *cookie*, you can do as following:
```
r = requests.get('https://t.me/v/?views=' + key, timeout=20, headers={
'x-requested-with':'XMLHttpRequest',
'user-agent':'Mozilla/5.0 (Windows NT 6.2; Win64; x64; rv:59.0) Gecko/20100101 Firefox/59.0',
'referer':'https://t.me/'+channel+'/'+post+'?embed=1',
'cookie':cookie}, proxies={'https':proxy}
)
```

**Comment by thangnc3** [comment]
> @jburagev,
> try:
> 
> ```
> r = requests.request("GET", url, params=querystring, proxies={'https':proxy})
> cookie = r.headers['set-cookie'].split(';')[0]
> key = r.text.split('data-view="')[1].split('"')[0]
> if 'stel_ssid' in cookie: 
>   return {'key':key,'cookie':cookie}
> else:
>   return False
> ```
> 
> after getting _key_ and _cookie_, you can do as following:
> 
> ```
> r = requests.get('https://t.me/v/?views=' + key, timeout=20, headers={
> 'x-requested-with':'XMLHttpRequest',
> 'user-agent':'Mozilla/5.0 (Windows NT 6.2; Win64; x64; rv:59.0) Gecko/20100101 Firefox/59.0',
> 'referer':'https://t.me/'+channel+'/'+post+'?embed=1',
> 'cookie':cookie}, proxies={'https':proxy}
> )
> ```

Dear skmohammadi,
I try use scrip but not working. Can u help me check it.

Error:

Traceback (most recent call last):
  File "D:/Projects/TeleBot/HackViewBot.py", line 50, in <module>
    thread = threading.Thread(target=run,args=(sys.argv[1],sys.argv[2],p))
IndexError: list index out of range
>>> 

Thank you!
[ViewBot.txt](https://github.com/MarkSnaile/telegram-channel-views-boost/files/3712015/ViewBot.txt)

**Comment by Dartrisen** [comment]
Ahah nice another shitty booster bot

Try to learn something about telebot first, kiddo

**Comment by mohammadyacn314** [comment]
syntax error is just for your python version 3 
downgrade v2
:/
Thread with proxy ... has been terminated. This error
how and which kind of proxy should set ??

**Comment by EliteDaMyth** [comment]
> @jburagev,
> try:
> 
> ```
> r = requests.request("GET", url, params=querystring, proxies={'https':proxy})
> cookie = r.headers['set-cookie'].split(';')[0]
> key = r.text.split('data-view="')[1].split('"')[0]
> if 'stel_ssid' in cookie: 
>   return {'key':key,'cookie':cookie}
> else:
>   return False
> ```
> 
> after getting _key_ and _cookie_, you can do as following:
> 
> ```
> r = requests.get('https://t.me/v/?views=' + key, timeout=20, headers={
> 'x-requested-with':'XMLHttpRequest',
> 'user-agent':'Mozilla/5.0 (Windows NT 6.2; Win64; x64; rv:59.0) Gecko/20100101 Firefox/59.0',
> 'referer':'https://t.me/'+channel+'/'+post+'?embed=1',
> 'cookie':cookie}, proxies={'https':proxy}
> )
> ```

i cant seem to get it to work any fixes?

**Comment by sinamcr7** [comment]
> @jburagev,
> try:
> 
> ```
> r = requests.request("GET", url, params=querystring, proxies={'https':proxy})
> cookie = r.headers['set-cookie'].split(';')[0]
> key = r.text.split('data-view="')[1].split('"')[0]
> if 'stel_ssid' in cookie: 
>   return {'key':key,'cookie':cookie}
> else:
>   return False
> ```
> 
> after getting _key_ and _cookie_, you can do as following:
> 
> ```
> r = requests.get('https://t.me/v/?views=' + key, timeout=20, headers={
> 'x-requested-with':'XMLHttpRequest',
> 'user-agent':'Mozilla/5.0 (Windows NT 6.2; Win64; x64; rv:59.0) Gecko/20100101 Firefox/59.0',
> 'referer':'https://t.me/'+channel+'/'+post+'?embed=1',
> 'cookie':cookie}, proxies={'https':proxy}
> )
> ```

thanks for your solution but it made me confused can you explain how it should be? if possible give me a contact way to converse easier

**Comment by jburagev** [comment]
> > @jburagev,
> > try:
> > ```
> > r = requests.request("GET", url, params=querystring, proxies={'https':proxy})
> > cookie = r.headers['set-cookie'].split(';')[0]
> > key = r.text.split('data-view="')[1].split('"')[0]
> > if 'stel_ssid' in cookie: 
> >   return {'key':key,'cookie':cookie}
> > else:
> >   return False
> > ```
> > 
> > 
> > after getting _key_ and _cookie_, you can do as following:
> > ```
> > r = requests.get('https://t.me/v/?views=' + key, timeout=20, headers={
> > 'x-requested-with':'XMLHttpRequest',
> > 'user-agent':'Mozilla/5.0 (Windows NT 6.2; Win64; x64; rv:59.0) Gecko/20100101 Firefox/59.0',
> > 'referer':'https://t.me/'+channel+'/'+post+'?embed=1',
> > 'cookie':cookie}, proxies={'https':proxy}
> > )
> > ```
> 
> thanks for your solution but it made me confused can you explain how it should be? if possible give me a contact way to converse easier

Write me on buragev1@gmail.com

**Comment by jburagev** [comment]
> @partizanfm88 did you fix the script? please tell me.

Write me on buragev1@gmail.com

**Comment by jburagev** [comment]
> syntax error is just for your python version 3
> downgrade v2
> :/
> Thread with proxy ... has been terminated. This error
> how and which kind of proxy should set ??

Write me on buragev1@gmail.com

**Comment by Mertua1** [comment]
Please can you edit the script and send it the correct one.  Please sir

**Comment by Mertua1** [comment]
> It seems with `url = https://t.me/v/?views= + key` in `addViewToPost` function, the problem will be fixed!

Please can you edit the script and send it the correct one. Please sir
